packages/edit-site/src/components/block-editor/use-site-editor-settings.js

```diff
@@ -157,7 +158,7 @@ export function useSpecificEditorSettings() {
 			hasFixedToolbar,
 			keepCaretInsideBlock,
 			defaultRenderingMode,
-
+			onSelectPost: useLink,
```

Comment:
I think there are a number of issues with this approach:
- `onSelectPost` doesn't sound like a react hook, which I think is a little dangerous. It means people might be encouraged to use it in a way that breaks the rule of hooks, or to use it outside of react components.
- The `onSelectPost` implementation in the post editor isn't a hook, so that inconsistency might be confusing to other devs.
- It adds a lot of overhead to the implementation in `usePostHistory`, which wasn't necessary before.
- The name `onSelectPost` is also misleading as it isn't used as an `onClick` handler, instead it's called to return an object with the `href` and the `onClick` handler.

So this needs some work. 🤔 

The way to solve this is going to be interesting. Some I can think of:

We could just pass a react hook in to the settings called something like `usePostLink`, and make the post editor work the same way. Not sure how @youknowriad feels about that 😄 .

Or you could do some refactoring in the site editor `useLink` hook so that the same can be achieved without using a hook, which doesn't seem to complicated.

In the long run, it seems it's not as simple as a single `onSelectPost` function if we need both the `href` and the `onClick` handler, it also probably needs to be renamed to `getPostLinkProps` or something.

Reply:
Thanks @talldan. Yeh, I had doubts about the approach also, but it at least proved it is possible to get things working in the site editor using the existing routing one way or another.

I have pushed [a very quick modification](https://github.com/WordPress/gutenberg/pull/57036/commits/ff77bebd69331d6d8643873f19c39943431c5af6) using your idea of switching to `getPostLinkProps` which makes more sense than `onSelect` with href added. I have also added a custom hook in the site editor to retrieve a version of `getPostLinkProps` that uses the routing. 

I have only given it a quick test and it seems to work, but will need some tidy up/refining if we want to go this way instead - I thought it was worth pushing it anyway so we can easily compare the two approaches.

Reply:
I'm not sure that I'm a big fan of passing a setting as a hook. I wonder why not just pass a regular callback that does "history.goTo" or something like that?

Reply:
Yeh, I refactored it with [this commit](https://github.com/WordPress/gutenberg/pull/57036/commits/ff77bebd69331d6d8643873f19c39943431c5af6) to pass a callback in both the site and post editor. In the site editor it is using the existing routing history mechanism and in the post editor the history array we implemented yesterday. A mentioned above this was a very quick refactor, it needs some refining.

Reply:
FYI - the reason for adding the `href` as well as the `onClick` was to match the `Edit` buttons in the post editor to those in the site editor which allow for a right click and open in new tab due to the presence of the href on the button, which we didn't have with the simple onSelectPost handler.